core/src/main/java/org/apache/iceberg/NoOpOperation.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+/**
+ * Performs a no-op commit.
+ *
+ * <p>This update is not exposed though the Table API. Instead, it is a package-private part of the
+ * Transaction API intended for use in {@link ManageSnapshots}.
+ */
+class NoOpOperation extends BaseValidatablePendingUpdate<NoOpOperation, Void>
+    implements ValidatablePendingUpdate<NoOpOperation, Void> {
+
+  private final String tableName;
+  private final TableOperations ops;
+
+  NoOpOperation(String tableName, TableOperations ops) {
+    this.tableName = tableName;
+    this.ops = ops;
+  }
+
+  @Override
+  public Void apply() {
+    return null;
+  }
+
+  @Override
+  public void commit() {
+    validateCurrent(SerializableTable.copyOf(new BaseTable(ops, tableName)));
```

Comment:
It doesn't seem worth adding table name to all of the operation constructors just to construct this `BaseTable`. What about just passing null? The table is only used for validation.

Also, I think it would be better to use a `BaseTable` with an underlying `StaticTableOperations`. You'll need to add a constructor that allows you to pass in the `TableMetadata`.

Reply:
> Also, I think it would be better to use a BaseTable with an underlying StaticTableOperations.

Great point, done. 

> It doesn't seem worth adding table name to all of the operation constructors just to construct this BaseTable. What about just passing null? The table is only used for validation.

Technically, it's possible for a user to access the table name inside a validation context e.g. 
```
.addValidation(
  currentTable -> Objects.equals(currentTable.name(), "abc"), 
  "Table name doesn't match our expected value of `abc`")
```
This validation will obviously fail if we pass in `null`. I realize this is an extremely unlikely use-case,  however since it's pretty easy for us to pass in the correct table name, I'd rather do the "right" thing than pass around `null` values. 

Those are just my thoughts. If you feel strongly about it, just let me know and I can make the change 👍